src/goToLocation.ts

```diff
@@ -25,7 +29,12 @@ export function gotoLocation(location: WindowLocation): void {
         )
         .pop()?.viewColumn || ViewColumn.Beside;
   }
-  workspace.openTextDocument(Uri.parse(location.uri)).then((textDocument) =>
+  const uri = Uri.parse(location.uri);
+  // vscode will cache the virtual documents even after closing unless onDidChange is fired
+  if (uri.scheme == "metalsDecode" && metalsFileProvider) {
+    metalsFileProvider.onDidChangeEmitter.fire(uri);
+  }
+  workspace.openTextDocument(uri).then((textDocument) =>
```

Comment:
I wonder why is that a bad thing?

Reply:
Because the output of the virtual docs can change.  If you run javap and then change the source and re-run it should refresh the output.  Also - if you run `Display Target` as Metals is indexing then it won't know the list of source jars yet so it will say that all jar dependencies are missing sources.  If you re-run after it's got the source jars then it will now show updated build info.

Reply:
got it, thanks